src/codewhisperer/util/commonUtil.ts

```diff
@@ -27,3 +32,43 @@ export function isAwsError(error: any): boolean {
         error.time instanceof Date
     )
 }
+
+export function getFileExt(languageId: string) {
+    switch (languageId) {
+        case 'java':
+            return '.java'
+        case 'python':
+            return '.py'
+        default:
+            break
+    }
+    return undefined
+}
+
+export async function getDirSize(
```

Comment:
I see that this is not part of fs-extra because it is complex: https://github.com/jprichardson/node-fs-extra/issues/656

- That issue mentions symlinks (which can be circular, causing an infinite loop).
    - Perhaps when recursing, we should pass a "depth" parameter and cancel recursion if depth exceeds e.g. 50. That avoids infinite loops. 
    - We could return a result that indicates if the depth limit is reached.
- This function is not specific to CW. So it should live in `src/shared/filesystemUtilities.ts`

Reply:
Thanks for the callout Justin.

- For symbolicLink, please check https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/fs.d.ts#L50 there's a `isSymbolicLink()` and we're already using it for Java dependency graph https://github.com/aws/aws-toolkit-vscode/blob/master/src/codewhisperer/util/dependencyGraph/javaDependencyGraph.ts#L307; in this util function, we only check `isDirectory ` and `isFile`, which I should've avoided circular concern;
- sure, I can move it to `filesystemUtilities `

Reply:
Also to add to the "depth", this function has time based stop condition, which the limit now is 10 seconds.